TACC/Core-CMS-Custom
# task/FP-1816-Extension-Request

## Overview

- Created form to submit extension request


## Related

- [FP-1816](https://jira.tacc.utexas.edu/browse/FP-1816)
- requires https://github.com/TACC/Core-CMS-Custom/pull/75

## Changes

- Modified APCD_Database utils to create a extension entry and a get gubmitter for extension and exception since they require the same information to be submitted to the DB with the form submission
- Created add/remove button to add extension time frame information for up to 5 extension requests
- Created path to be submissions/exception-request

## Testing

1. Go to http://localhost:8000/submissions/extension-request/
2. Use snippet on line 20 of extension/views.py making sure to put in your CMS username in the placeholder text
    <details><summary>Snippet</summary>

         [(1, "TESTGOLD", 10000000, "yourusernamehere")]

     </details>
3.  Make sure you have to abide by helper text directions for input on fields
4. Dates can be in past - no limit here since there's no specification from UT Health at this point
5. Should get Credentials Required error that creates a ticket.


## UI
<img width="1382" alt="Screen Shot 2023-01-27 at 10 51 59 AM" src="https://user-images.githubusercontent.com/96220951/215145919-c62ab9bc-2e4b-4345-b4bf-c64f92800cd1.png">

<img width="1419" alt="Screen Shot 2023-01-27 at 10 52 26 AM" src="https://user-images.githubusercontent.com/96220951/215146027-c2fd1b58-8d0c-4c1d-8f1c-f307ddf08427.png">

<img width="1410" alt="Screen Shot 2023-01-27 at 10 52 48 AM" src="https://user-images.githubusercontent.com/96220951/215146102-bc478146-b7f2-4788-8e9e-66ed264e1f43.png">
<img width="1482" alt="Screen Shot 2023-01-27 at 10 53 16 AM" src="https://user-images.githubusercontent.com/96220951/215146231-2a06eeb8-9e12-4cc3-9b55-75b8de5426cd.png">

## Notes
Need this on staging to test if, once it's past the credentials check, that it adds input from the add extension button fields
